connector/connect/server/src/main/scala/org/apache/spark/sql/connect/utils/ErrorUtils.scala

```diff
@@ -57,28 +69,105 @@ private[connect] object ErrorUtils extends Logging {
     classes.toSeq
   }
 
-  private def buildStatusFromThrowable(st: Throwable, stackTraceEnabled: Boolean): RPCStatus = {
+  private def serializeClasses(t: Throwable): String = {
+    JsonMethods.compact(JsonMethods.render(allClasses(t.getClass).map(_.getName)))
+  }
+
+  private[connect] val NUM_ERRORS_LIMIT = 5
+
+  // We can get full exception messages and optionally stacktrace by
+  // a separate RPC call if enrichErrorEnabled is true. So imposing a smaller
+  // limit to reduce the probability of hitting the 8KB header limit.
+  private val MAX_MESSAGE_SIZE = 512
+
+  // Convert Throwable to a protobuf message FetchErrorDetailsResponse.
+  // Truncate error messages by default.
+  private[connect] def throwableToFetchErrorDetailsResponse(
```

Comment:
This logic gets unnecessarily complex because the implementation now has to deal with two cases, injected into the response and explicit RPC. If we keep it just as the explicit RPC, the extra branches are not needed.

Reply:
I will remove it then...